zha/application/platforms/sensor/__init__.py

```diff
@@ -1947,3 +1949,14 @@ class DanfossMotorStepCounter(Sensor):
     _attribute_name = "motor_step_counter"
     _attr_translation_key: str = "motor_stepcount"
     _attr_entity_category = EntityCategory.DIAGNOSTIC
+
+
+@MULTI_MATCH(cluster_handler_names=CLUSTER_HANDLER_ELECTRICAL_CONDUCTIVITY)
+class ElectricalConductivity(Sensor):
+    """Electrical Conductivity sensor."""
+
+    _attribute_name = "measured_value"
+    _attr_device_class: SensorDeviceClass = SensorDeviceClass.CONDUCTIVITY
+    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
+    _attr_translation_key: str = "electrical_conductivity"
+    _attr_native_unit_of_measurement = ELECTRICAL_CONDUCTIVITY_DECISIEMENS_PER_METER
```

Comment:
As mentioned in the other comment, `UnitOfConductivity.MILLISIEMENS_PER_CM` should be used here then.

Reply:
`UnitOfConductivity.MICROSIEMENS_PER_CM` seems to be the unit commonly used for soil electrical conductivity and that matches what Z2M does.